data-jdbc/src/test/java/io/micronaut/data/jdbc/h2/H2BookRepository.java

```diff
@@ -31,6 +32,7 @@
 import java.util.Set;
 
 @JdbcRepository(dialect = Dialect.H2)
+@Join(value = "genre", type = Join.Type.LEFT_FETCH)
```

Comment:
Please do not add the join for all the methods. Some methods might test getting the entity without any joins.
Please add an extra method for findOne/All for your test case, you can use different prefixes like query, list etc.

Reply:
Makes a lot of sense, fixed it. Thanks